backend/src/services/api.rs

```diff
@@ -441,3 +441,101 @@ pub async fn raw_configuration_by_node(
         .await?,
     ))
 }
+
+/// Get Detailed Information on Node
+///
+/// # Errors
+///
+/// This function will return an error if the server was unable to get node'a client
+/// or one of the requests to get information from the node fails
+#[cfg_attr(feature = "swagger", utoipa::path(
+        get,
+        context_path = ApiV1::to_path(),
+        path = "/nodes/{node_name}/detailed",
+        params (
+            ("id", description = "Node's ID")
+        ),
+        responses(
+            (status = 200, body = DetailedNode, content_type = "application/json", description = "Detailed Node information"),
+            (status = 401, description = "Unauthorized"),
+            (status = 404, description = "Node Not Found")
+        ),
+        security(("api_key" = []))
+    ))]
+pub async fn get_detailed_node_info(
+    Extension(client): Extension<HttpBobClient>,
+    Path(node_name): Path<NodeName>,
+) -> AxumResult<Json<DetailedNode>> {
+    tracing::info!("get /nodes/{node_name}/detailed : {client:?}");
+    let handle = Arc::new(
+        client
+            .api_secondary(&node_name)
+            .cloned()
+            .ok_or(StatusCode::NOT_FOUND)?,
+    );
+
+    let status = {
+        let handle = handle.clone();
+        tokio::spawn(async move { handle.get_status().await })
+    };
+    let metrics = {
+        let handle = handle.clone();
+        tokio::spawn(async move { handle.clone().get_metrics().await })
+    };
+    let space_info = {
+        let handle = handle.clone();
+        tokio::spawn(async move { handle.clone().get_space_info().await })
+    };
+    let disks = {
+        let handle = handle.clone();
+        tokio::spawn(async move { handle.clone().get_disks().await })
+    };
+
+    let Ok(Ok(GetStatusResponse::AJSONWithNodeInfo(status))) = status.await else {
```

Comment:
Am I correct that tasks for `metrics`, `space_info` and `disks` will still run to completion even if function shortcircuits here? Wouldn't this approach result in many tasks running for every unsuccessful request?